src/ooxml/java/org/apache/poi/xwpf/usermodel/XWPFDocument.java

```diff
@@ -1603,9 +1603,69 @@ public XWPFTableCell getTableCell(CTTc cell) {
         }
         return tableRow.getTableCell(cell);
     }
-
+    
     @Override
     public XWPFDocument getXWPFDocument() {
         return this;
     }
+    
+    /**
+     * This method is used to create template for chart XML
+     * no need to read MS-Word file and modify charts
+     * @return This method return object of XWPFChart Object with default height and width
+     * @throws InvalidFormatException
+     * @throws IOException
+     * @since POI 4.0.0
+     */
+    public XWPFChart addChart() throws InvalidFormatException, IOException{
+        return addChart(XWPFChart.DEFAULT_WIDTH, XWPFChart.DEFAULT_HEIGHT);
+    }
+    
+    /**
+     * This method is used to create template for chart XML
+     * no need to read MS-Word file and modify charts
+     * @param width width of chart
+     * @param height height of chart
+     * @return This method return object of XWPFChart
+     * @throws InvalidFormatException
+     * @throws IOException
+     * @since POI 4.0.0
+     */
+    public XWPFChart addChart(int width, int height) throws InvalidFormatException, IOException{
+        
+        //get chart number
+        int chartNumber = getPackagePart().getPackage().
+                getPartsByContentType(XWPFRelation.CHART.getContentType()).size() + 1;
+        
+        //create relationship in document for new chart
+        RelationPart rp = createRelationship(
+                XWPFRelation.CHART, XWPFFactory.getInstance(), chartNumber, false);
+        
+        //get chart relationship id
+        String chartId = rp.getRelationship().getId();
+        
+        //create paragraph and run object
+        XWPFRun xRun = this.createParagraph().createRun();
+        
+        CTInline inline = xRun.addChart(width, height,chartId);
+        
+        //get package part of xwpfchart object
+        XWPFChart xwpfChart=rp.getDocumentPart();
+        
+        //create embedded part for embedded xlsx file
+        RelationPart xlsx = xwpfChart.createRelationship(XWPFRelation.CHART_SHEET,chartNumber);
+        
+        //add output stream in xwpfchart object
+        xwpfChart.addEmbeddedWorkSheet(xlsx.getDocumentPart().getPackagePart().getOutputStream());
+        
+        //set in line object into xwpfchart object
+        xwpfChart.setInLine(inline);
```

Comment:
Maybe, instead of this `@Internal` method, you could define a public `attachTo` method accepting the `XWPFRun` as parameter, along with the boundaries and chart I'd.

Reply:
Will update in next checkin